packages/terra-time-input/src/TimeInput.jsx

```diff
@@ -533,8 +586,48 @@ class TimeInput extends React.Component {
     if ((event.key === '=' || event.key === '+') && !this.props.atMaxDate) {
       const currentTimeValue = this.formatHour(hour, meridiem).concat(':', minute).concat(this.props.showSeconds ? ':'.concat(second) : '');
       if (TimeUtil.validateTime(currentTimeValue, this.props.showSeconds)) {
-        if (minute === '59') {
-          minute = '00';
+        if (inputType === 3 && this.props.showSeconds) {
+          if (second === '59') {
+            second = '00';
+            if (minute === '59') {
+              minute = '00';
+              if (hour === '23') {
+                hour = '00';
+              } else {
+                if (variant === TimeUtil.FORMAT_12_HOUR && hour === '11') {
+                  if (meridiem === this.anteMeridiem) {
+                    meridiem = this.postMeridiem;
+                  } else {
+                    meridiem = this.anteMeridiem;
+                  }
+                }
+                hour = TimeUtil.incrementHour(hour, variant);
+              }
+            } else {
+              minute = TimeUtil.incrementMinute(minute);
+            }
+          } else {
+            second = TimeUtil.incrementSecond(second);
+          }
+        } else if (inputType === 1) {
+          if (minute === '59') {
+            minute = '00';
+            if (hour === '23') {
+              hour = '00';
```

Comment:
what does zero represents in hour field

Reply:
In 24 hr format, 12 am is rep as 00 right?

Reply:
Yes, in 24 hr format, 12 am is represents 00
<img width="317" alt="Screenshot 2024-01-23 at 2 33 53 PM" src="https://github.com/cerner/terra-framework/assets/127083404/10797b17-e8b0-478f-bdd5-203759c06a17">